### PR TITLE
Creates an EMP act for Jetpacks + minor buffs to sec modsuits

### PR DESCRIPTION
**Pull request by EnterTheJake**
## About The Pull Request

Jetpacks are now vulnerable to EMP, they get disabled for a few seconds after getting hit by an EMP blast.

Sec modsuits now start with the jetpack module preinstalled instead of the mirage module,  and their starting powercell has been upgraded.

## Why It's Good For The Game

It's no secret that regular jetpacks and your run of the mill eva suit see far more use than actual security issued equipment.

Having your best option available from the start of the shift is just boring, and doesn't encourage the crew to even engage with Modsuits in any way.

With this PR i'm introducing a weakness to jetties by having em turn of for 4 seconds if you get hit by a pulse.

In exchange sec mods are now a  bit more suited for their job, Mirage module has been swapped with Jetpack module and they now spawn with a better cell.

## Changelog

:cl:

balance: Jetpacks are now briefly disabled by EMP.
balance: Mirage module has been removed from Sec modsuits.
balance: Security modsuits now spawn with a better cell and the jetpack module preinstalled.
/:cl:

### CODE REVIEW

**Comment by zoomachina** [comment]
You give up a backpack slot to use a jetpack from EVA storage, if modsuit jetpacks are somehow worse than normal ones it's because they're really weak

**Comment by EnterTheJake** [comment]
> You give up a backpack slot to use a jetpack from EVA storage, if modsuit jetpacks are somehow worse than normal ones it's because they're really weak

Insignificant, you can still carry a bag around with you.

 It's annoyng but more than makes up not having to deal with all the tradeoffs of wearing a modsuit.
 
 And i do agree they are weak, that's why i'm buffing em.

**Comment by DaCoolBoss** [comment]
Randomise the movement + slowdown instead of forcing a full stop, because 4 seconds of them being unable to move without throwing items makes it trivial to combo EMP + any ranged weapon to instawin space combats. Especially with an EMP proof modsuit and ballistic gun. 

Also not sure about making this affect old school jetpacks at all? Losing bag slot is a huge deal, bag needs to be dropped or stashed to use a 2 handed gun. Doesn't really match the flavour either, jetpacks don't use electronics besides pressure readouts on some sprites.

**Comment by moocowswag** [comment]
Does this only effect electrical/ion  jetpacks 
and not mechanical ones that use gasses?

**Comment by EnterTheJake** [comment]
> Randomise the movement + slowdown instead of forcing a full stop, because 4 seconds of them being unable to move without throwing items makes it trivial to combo EMP + any ranged weapon to instawin space combats. Especially with an EMP proof modsuit and ballistic gun.
> 
> Also not sure about making this affect old school jetpacks at all? Losing bag slot is a huge deal, bag needs to be dropped or stashed to use a 2 handed gun. Doesn't really match the flavour either, jetpacks don't use electronics besides pressure readouts on some sprites.



> Does this only effect electrical/ion jetpacks and not mechanical ones that use gasses? (The makeshift one for example)
> 
> if you are flying through space with a jetpack and you get hit by emp do you continue going through space at the same or similar speed or do you just suddenly stop?

It effects all jetpacks, upon being hit you root for 4 seconds but can still change z level.

**Comment by moocowswag** [comment]
I dont think it should delete your velocity if youre on a space turf atleast

**Comment by EnterTheJake** [comment]
> Randomise the movement + slowdown instead of forcing a full stop, because 4 seconds of them being unable to move without throwing items makes it trivial to combo EMP + any ranged weapon to instawin space combats. Especially with an EMP proof modsuit and ballistic gun.
> 
> Also not sure about making this affect old school jetpacks at all? Losing bag slot is a huge deal, bag needs to be dropped or stashed to use a 2 handed gun. Doesn't really match the flavour either, jetpacks don't use electronics besides pressure readouts on some sprites.

No, that's entirely the point, a modsuit should be better in space  than an eva suit with a jetpack.

**Comment by zoomachina** [comment]
> No, that's entirely the point, a modsuit should be better in space than an eva suit with a jetpack.

1. Why?
2. Modsuit hit by EMP might suffer even worse problems
3. If you really think that, why won't you make modsuit jetpack better instead of making normal jetpack worse and calling it a buff to the former?

**Comment by EnterTheJake** [comment]
> > No, that's entirely the point, a modsuit should be better in space than an eva suit with a jetpack.
> 
> 1. Why?
> 2. Modsuit hit by EMP might suffer even worse problems
> 3. If you really think that, why won't you make modsuit jetpack better instead of making normal jetpack worse and calling it a buff to the former?



> > No, that's entirely the point, a modsuit should be better in space than an eva suit with a jetpack.
> 
> 1. Why?
> 2. Modsuit hit by EMP might suffer even worse problems
> 3. If you really think that, why won't you make modsuit jetpack better instead of making normal jetpack worse and calling it a buff to the former?



> > No, that's entirely the point, a modsuit should be better in space than an eva suit with a jetpack.
> 
> 1. Why?
> 2. Modsuit hit by EMP might suffer even worse problems
> 3. If you really think that, why won't you make modsuit jetpack better instead of making normal jetpack worse and calling it a buff to the former?

1) Because modsuits are specifically designed to be the go to option for going to space, having to carry a bag it's a minor inconvenience when you don't need to manage your power, modules or the suit slowdown.
2) Modsuits emp act is fairly forgiving, it can randomly pulse a few wires but that's about it, besides with the tech rework you can gain access to emp shielding in the first 10 minutes of the shift, go to robotics and ask for an upgrade.
3) There is no need, the jetpack module is good as it is, maybe a bit power hungry, but that's about it.

**Comment by zoomachina** [comment]
> 1. Because modsuits are specifically designed to be the go to option for going to space, having to carry a bag it's a minor inconvenience when you don't need to manage your power, modules or the suit slowdown.

Objectively wrong, modsuits are specifically designed to be modular, i.e. being useful for a variety of purposes depending on the module set, whereas jetpacks' only purpose is to be used for spacefaring (and even then they take up a bag slot, even if you refuse to acknowledge it's a major drawback). I think it makes sense for the jack-of-all-trades option to be weaker than specialist one. 
Furthermore, both security modsuits and jetpacks are available roundstart, so 
> Having your best option available from the start of the shift is just boring (c) "Why It's Good For The Game"

makes no sense in the context of justifying your pr.
> 3. There is no need, the jetpack module is good as it is, maybe a bit power hungry, but that's about it.

It's also slower than a normal jetpack (as it should be)

**Comment by Higgin** [comment]
Secmod having a jetpack stock is a good change  Unfortunately, free space tiding is still a thing, in spite ending that being one of the goals of MODsuits, and they're overdue for what should rightly be considered a piece of necessary kit.

**Comment by SomeoneYouProbablyKnow** [comment]
"Minor buffs to sec"
look inside
Doubles the amount of jetpacks easily accessible to security

Considering how stupid jetpack speed is and how often nuckies disable grav gen this is a crazy buff.
It's not like a jetpack is only useful outside the station and the minor emp effect does nout outweigh the buff that makes sec even more self sufficient.

**Comment by moocowswag** [comment]
I dont think sec modsuits should have better cells for the sake of having better cells, they should already be encouraged to get better cells and by extension, get their gun rechargers upgraded, probably the best excuse for warden to be out of brig.

**Comment by EnterTheJake** [comment]
@zoomachina 

Modsuits are designed to be modular with the ability to specialize in one area.

 Sec modsuits were specifically made to hunt criminals in space, as there's not much of a reason to ever use em on station.

Both jetties and mods are available round start, modsuits however need to be mantained and upgraded, jetties have no need of such a thing.

 As you've described Jetpacks are already faster than the modules and become even more desired once adaptations or hypernob crystals have been obtained as you don't even need to wear an eva suit anymore.

**Comment by moocowswag** [comment]
> Sec modsuits were specifically made to hunt criminals in space, as there's not much of a reason to ever use em on station.

the pepper spray defense module , which is useless against someone with eva gear

**Comment by SomeoneYouProbablyKnow** [comment]
> Sec modsuits were specifically made to hunt criminals in space, as there's not much of a reason to ever use em on station.

There would be if they had jetpacks. 

And yes you can take the limited eva jetpacks or order them or craft improvised jetpacks but doing any of that is part of the game. Security shouldn't get everything handed to them because "their job is to hunt valids"

**Comment by EnterTheJake** [comment]
> > Sec modsuits were specifically made to hunt criminals in space, as there's not much of a reason to ever use em on station.
> 
> the pepper spray defense module , which is useless against someone with eva gear, but thats the only contradiction i think



> > Sec modsuits were specifically made to hunt criminals in space, as there's not much of a reason to ever use em on station.
> 
> the pepper spray defense module , which is useless against someone with eva gear, but thats the only contradiction i think



> > Sec modsuits were specifically made to hunt criminals in space, as there's not much of a reason to ever use em on station.
> 
> There would be if they had jetpacks.
> 
> And yes you can take the limited eva jetpacks or order them or craft improvised jetpacks but doing any of that is part of the game. Security shouldn't get everything handed to them because "their job is to hunt valids"

The idea was for sec to go to robotics to requested a module, admittedly they should probably have had something so crucial to their job at roundstart.

**Comment by DaydreamIQ** [comment]
> "Minor buffs to sec" look inside Doubles the amount of jetpacks easily accessible to security
> 
> Considering how stupid jetpack speed is and how often nuckies disable grav gen this is a crazy buff. It's not like a jetpack is only useful outside the station and the minor emp effect does not outweigh the buff that makes sec even more self sufficient.
> 
> The emp effect is neat, not TOO debilitating but sufficiently annoying

This wont affect nukies in the slightest. Sec get ion jetpacks which are considerably slower, Nukies get advanced ones which have the same speed as EVA jetpacks. One of the most annoying things to deal with as sec is someone in space because you literally cannot catch up to anyone with a jetpack which leads to really awkward baton jousting that you're never going to be able to capitalize on

**Comment by SimplySmiley123** [comment]
> > Randomise the movement + slowdown instead of forcing a full stop, because 4 seconds of them being unable to move without throwing items makes it trivial to combo EMP + any ranged weapon to instawin space combats. Especially with an EMP proof modsuit and ballistic gun.
> > Also not sure about making this affect old school jetpacks at all? Losing bag slot is a huge deal, bag needs to be dropped or stashed to use a 2 handed gun. Doesn't really match the flavour either, jetpacks don't use electronics besides pressure readouts on some sprites.
> 
> No, that's entirely the point, a modsuit should be better in space than an eva suit with a jetpack.

And this does not fix a damn thing about that but make it overall much more annoying for everyone.
People would still pick regular jetpack and space suit over the hassle of a modsuit anyway.

**Comment by SimplySmiley123** [comment]
> "Minor buffs to sec" look inside Doubles the amount of jetpacks easily accessible to security
> 
> Considering how stupid jetpack speed is and how often nuckies disable grav gen this is a crazy buff. It's not like a jetpack is only useful outside the station and the minor emp effect does not outweigh the buff that makes sec even more self sufficient.
> 
> The emp effect is neat, not TOO debilitating but sufficiently annoying

Modsuit jetpacks are slow as hell, and the EMP effect is a killer, 4 seconds with no momentum preservation sucks hard with how easy it is to get EMPs.

**Comment by SimplySmiley123** [comment]
> 
> 1. Because modsuits are specifically designed to be the go to option for going to space, having to carry a bag it's a minor inconvenience when you don't need to manage your power, modules or the suit slowdown.
> 2. Modsuits emp act is fairly forgiving, it can randomly pulse a few wires but that's about it, besides with the tech rework you can gain access to emp shielding in the first 10 minutes of the shift, go to robotics and ask for an upgrade.
> 3. There is no need, the jetpack module is good as it is, maybe a bit power hungry, but that's about it.

1) Supposed to be, but the way they are now no one wants to use them because of how big of  a pain it is compared to literally any other EVA methods.
2) It's not forgiving, you got an RNG roll of a dice deciding wether or not you'll be able to take your suit off at all or interact with it without shocking yourself.
3) It's not, it's slow as hell and power hungry, you're better off with a makeshift jetpack then the jetpack module.

**Comment by SomeoneYouProbablyKnow** [comment]
> This wont affect nukies in the slightest. Sec get ion jetpacks which are considerably slower, Nukies get advanced ones which have the same speed as EVA jetpacks. 

I'd much rather have that than not. That way i'd have at least have a chance. So it's not insignificant especially considering how scarce jetpacks are to begin with (Yes there are ways to get more but no one ever does)

> One of the most annoying things to deal with as sec is someone in space because you literally cannot catch up to anyone with a jetpack which leads to really awkward baton jousting that you're never going to be able to capitalize on

If the antag goes out of his way to acquire a jetpack, they deserve to run circles around sec at least untill they get their own. Don't want the heat, don't follow them into space.
You're not entitled to having all the countermeasures in brig ready for you, security is not 'easy hunt antags' mode.
Running to eva for a jetpack to get a perp in space is litererally THE game.

But yes jetpack speed combat is awful. Wish there was less of it not more

**Comment by Jackal-boop** [comment]
These changes should be split into distinct PRs

**Comment by Blacklist897** [comment]
remove advanced ions and make all forms of jetpack (apart from the jet harness and the modsuit fitting captains pack) the same speed

**Comment by CRITAWAKETS** [comment]
This currently makes the EMP effect work on the captain's jetpack and the syndicate jet harness. Probably shouldn't do that.

**Comment by Exester509** [comment]
I'm pretty sure the mirage module, as well as the pepper shoulders, cannot be printed from R&D. This would make them unobtainable.

Aditionally, it makes zero sense that an EMP can affect manual valves.

**Review by projectkepler-ru — COMMENTED** [review]
If you seriously want to buff sec modsuit I'd advise making them able to put the old security oxygen jetpack onto their suit slot and putting the jetpack into those bulky suit storage that doesnt even recharge the cell
Or give them a roundstart tether, an outright jetpack is rediculous. Yes, a lot of antag has space capability or rapid movement in space, but security make up for it is their numbers, coordination and they're generally allowed to use equipment freely

**Comment by EnterTheJake** [comment]
> I'm pretty sure the mirage module, as well as the pepper shoulders, cannot be printed from R&D. This would make them unobtainable.
> 
> Aditionally, it makes zero sense that an EMP can affect manual valves.

They still have electronics in them.

**Comment by Skebulba** [comment]
Weird for sec to get roundstart jetpacks in their suits but not engi imo

**Comment by alien3301** [comment]
The EMP effect on jetpacks is fine. Just move around environment stuff if enemies have emp.

**Comment by alien3301** [comment]
> Weird for sec to get roundstart jetpacks in their suits but not engi imo

Engi don't need those as much as sec. The modsuit is for pressure protection during breaches and setting up solars. In both cases you 99% of the time have grav or stuff to push off of.

File: code/game/objects/items/tanks/jetpack.dm [hunk 1]
```diff
@@ -8,9 +8,17 @@
 	w_class = WEIGHT_CLASS_BULKY
 	distribute_pressure = ONE_ATMOSPHERE * O2STANDARD
 	actions_types = list(/datum/action/item_action/set_internals, /datum/action/item_action/toggle_jetpack, /datum/action/item_action/jetpack_stabilization)
+	/// What gas our jetpack is filled with on initialize
 	var/gas_type = /datum/gas/oxygen
+	/// If the jetpack is currently active
 	var/on = FALSE
+	/// If the jetpack will have a speedboost in space/nograv or not
+	var/full_speed = TRUE
```
Comment (Fikou) [inline code/game/objects/items/tanks/jetpack.dm:16]: this has been removed

**Review by Fikou — CHANGES_REQUESTED** [review]
put mirage grenade dispensers into rnd please

**Pull request merged**